CMakeLists.txt

```diff
@@ -14,8 +14,8 @@ project(${SOLUTION_NAME} LANGUAGES CXX VERSION ${SOLUTION_NAME_VERSION})
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_BUILD_TYPE Debug)
-set(CMAKE_CXX_FLAGS_DEBUG "-O1 -Wall -Wextra")
-set(CMAKE_CXX_FLAGS_RELEASE "-O3 -Wall -Wextra")
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O1 -Wall -Wextra")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -Wall -Wextra")
```

Comment:
In the last two commits I looks like the debug build timed out without -01. I'll create and document a separate issue for optimization settings to prevent this from blocking things

Reply:
What I'd suggest for forcing the `-O1` in Travis is doing it on the `cmake` command line of the Travis job (e.g., `cmake -DCMAKE_CXX_FLAGS="-O1" ...` on `.travis/install.sh`, together with a comment) instead of hardcoding it in the `CMakeLists.txt` file, which will impose the flag on anyone building the software with a `Debug` release type.